src/payment/checkout/payment-form/flex-microform/FlexMicroform.jsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { setPaymentTokenFetch } from '../../../payment-methods/cybersource/service';
+
+class FlexMicroform extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            microform: null,
+        }
+        this.onLoad = this.onLoad.bind(this);
+        this.getPaymentToken = this.getPaymentToken.bind(this);
+    }
+
+    componentDidMount() {
+        setPaymentTokenFetch(this.getPaymentToken);
+
+        const script = document.createElement('script');
+        script.src = "https://flex.cybersource.com/cybersource/assets/microform/0.11/flex-microform.min.js";
+        script.async = true;
+
+        script.addEventListener('load', this.onLoad);
+        script.addEventListener('error', e => {
+            console.error(e);
+        });
+
+        document.body.appendChild(script);
+    }
+
+    initialize() {
+        if (this.state.microform !== null) {
+            return;
+        }
+        if (!this.props.captureKeyId) {
+            return;
+        }
+        if (window['Flex'] === undefined) {
+            return;
+        }
+        const microform = new window.Flex(this.props.captureKeyId).microform({
+            styles: { 
+                'input': {    
+                    'font-size': '16px',
+                    'font-family': 'Roboto, "Helvetica Neue", Arial, sans-serif',
+                    'color': '#2d323e',
+                },
+                ':disabled': {
+                    'cursor': 'not-allowed'
+                }
+            }
+        });
+        this.setState({microform: microform});
+    }
+
+    getPaymentToken(cardExpirationMonth, cardExpirationYear) {
+        const microformOptions = {    
+            cardExpirationMonth: cardExpirationMonth,  
+            cardExpirationYear: cardExpirationYear,
+        };
+        return new Promise((resolve, reject) => {
+            if (this.state.microform === null) {
+                reject('Microform not initialized');
+            }
+            this.state.microform.createToken(microformOptions, function (err, token) {
+                if (err) {
+                    reject(err);
+                } else {
+                    resolve(token);
+                }
+            });
+        });
+    }
+
+    componentDidUpdate() {
+        this.initialize();
```

Comment:
It's worth noting that componentDidUpdate will occur anytime the component rerenders, which may be more often than you want to call `initialize`.  

If anything, you may consider using the `componentDidUpdate(prevProps, prevState)` form of the method, which will let you compare captureKeyId to its previous value and choose to initialize _only_ when it has changed.  That way if you add more props to the component, you won't run the risk of blowing away your `microform` instance over and over.

Reply:
ou - prevProps is the way to go!